cgi/mapcache.c

```diff
@@ -297,6 +348,8 @@ int main(int argc, const char **argv)
       goto cleanup;
     }
 
+    set_headers(ctx, environ);
```

Comment:
`set_headers()` changes `environ` contents (thanks to `apr_strtok()` modifying its input string). I would use `apr_pstrdup()` to create a copy: `set_headers(ctx, apr_pstrdup(ctx->pool, environ));`

Reply:
Thanks for the explanation - updated

Reply:
See comment below - I was getting server crashes when making a full copy of the environ.

Reply:
My bad... `environ` is an array of strings, not a single string that `apr_pstrdup()` is designed to handle.